openeo/extra/job_management.py

```diff
@@ -472,18 +512,14 @@ def ignore_connection_errors(context: Optional[str] = None):
         time.sleep(5)
 
 
-class _JobDatabaseInterface:
-    def read(self) -> pd.DataFrame:
-        raise NotImplementedError()
-
-    def persist(self, df: pd.DataFrame):
-        raise NotImplementedError()
-
 
-class _CsvJobDatabase(_JobDatabaseInterface):
+class CsvJobDatabase(JobDatabaseInterface):
     def __init__(self, path: Union[str, Path]):
         self.path = Path(path)
 
+    def exists(self) -> bool:
+        return self.path.exists()
```

Comment:
add `is_file()` too

Reply:
reconsidering this: if it exists and it's not a file, things should fail at "read" time (as early as possible), not at "persist" time (when valuable work might have been gone to waste). Keeping as-is mean former behavior, adding `is_file()` changes to latter behavior.